### PR TITLE
GH-203 Fix SecurityInfo REST endpoint

### PR DESCRIPTION
**Pull request by ilayaperumalg**
- Use `SecurityProperties` instead of infering from environment
 - Thanks to @jvalkeal for the suggestion

This resolves spring-cloud/spring-cloud-dataflow/issues/203

### CODE REVIEW

**Comment by ilayaperumalg** [comment]
This fixes https://github.com/spring-cloud/spring-cloud-dataflow/issues/203

File: spring-cloud-dataflow-admin-starter/src/main/java/org/springframework/cloud/dataflow/admin/controller/SecurityController.java [hunk 1]
```diff
@@ -31,10 +32,11 @@
  */
 @RestController
 @RequestMapping("/security/info")
+@EnableConfigurationProperties(SecurityProperties.class)
 public class SecurityController {
 
 	@Autowired
-	private Environment environment;
+	SecurityProperties securityProperties;
```
Comment (markfisher) [inline spring-cloud-dataflow-admin-starter/src/main/java/org/springframework/cloud/dataflow/admin/controller/SecurityController.java]: why is this not private? also, can we go ahead and switch away from field injection while we're making changes here?

Comment (ilayaperumalg) [inline spring-cloud-dataflow-admin-starter/src/main/java/org/springframework/cloud/dataflow/admin/controller/SecurityController.java]: Sure, this needs to be private. I can also change it to use constructor level injection. Earlier I followed the typical approach to autowire the configuration properties via field injection.

**Comment by jvalkeal** [comment]
This should take away that previous error. `security.basic.enabled` is `true` by default so I don't know if application actually works if that is not disabled on admin(you'd need to find generated random password from logs). Local does disable it in its yml config. But that is another topic so I think this looks good. I anyway have `servers.yml` file which user can use to configure custom settings.

**Comment by jvalkeal** [comment]
Merged

**Comment by mbogoevici** [comment]
A bit late into the game, but we could add a `@PropertySource("adminDefaults.properties")` with defaults to the admin server to override Boot's own defaults. which can include disabling security. So that we don't have to do this for each admin server implementation.

**Comment by ilayaperumalg** [comment]
@mbogoevici I don't think we do something to override the existing `boot` properties here. All we do here is to expose whatever the settings for security property.

**Pull request closed without merging**